state/azure/cosmosdb/cosmosdb.go

```diff
@@ -377,95 +346,96 @@ func (c *StateStore) Delete(req *state.DeleteRequest) error {
 
 // Multi performs a transactional operation. succeeds only if all operations succeed, and fails if one or more operations fail.
 func (c *StateStore) Multi(request *state.TransactionalStateRequest) error {
-	operations := []CosmosOperation{}
+	if len(request.Operations) == 0 {
+		c.logger.Debugf("No Operations Provided")
+		return nil
+	}
+	partitionKey := request.Metadata["partitionKey"]
```

Comment:
You should use `populatePartitionMetadata(key, request.Metadata)`

That way it will use the item key if the `partitionKey` isn't explicitly defined.